corehq/apps/translations/integrations/transifex/client.py

```diff
@@ -138,9 +138,13 @@ def upload_resource(self, path_to_po_file, resource_slug, resource_name, update_
         :param resource_name: resource name, mostly same as resource slug itself
         :param update_resource: update resource
         """
+        create_missing_resource = False
         if update_resource:
-            resource = self._get_resource(resource_slug)
-        else:
+            try:
+                resource = self._get_resource(resource_slug)
+            except TransifexApiException:
+                create_missing_resource = True
+        if not update_resource or create_missing_resource:
```

Comment:
Out of curiosity, what's the situation where we're trying to update a resource but it doesn't exist? Does renaming something in the app do that?

Reply:
Kiran pointed out that it was unintuitive for the update function to only update existing resources and for the user to then have to submit another form to create newly created menus/forms as resources so this where that check comes from. The update function will now also delete resources of menus/forms that have been deleted on the HQ side.